test/metabase/api/pulse_test.clj

```diff
@@ -1107,6 +1109,61 @@
                   :recipient-type nil}
                  (mt/summarize-multipart-single-email (-> channel-messages :channel/email first) #"Daily Sad Toucans"))))))))
 
+(defmulti native-array-query
+  {:arglists '([driver])}
+  tx/dispatch-on-driver-with-test-extensions
+  :hierarchy #'driver/hierarchy)
+
+(defmethod native-array-query :default
+  [_driver]
+  "select array['a', 'b', 'c']")
+
+(doseq [driver [:redshift :databricks]]
+  (defmethod native-array-query driver
+    [_driver]
+    "select array('a', 'b', 'c')"))
+
+(doseq [driver [:mysql :sqlite]]
+  (defmethod native-array-query driver
+    [_driver]
+    "select json_array('a', 'b', 'c')"))
+
+(defmethod native-array-query :snowflake
+  [_driver]
+  "select array_construct('a', 'b', 'c')")
+
+(doseq [driver [:postgres :mysql :snowflake :databricks :redshift :sqlite :vertica]]
+  (defmethod driver/database-supports? [driver :test/arrays]
```

Comment:
Since this feature is meant to be used elsewhere I think the feature should go in `metabase.test.data.interface`   and driver specifics in their test.data namespace otherwise this ns needs to be loaded before `cache_test` and add to the driver.changelog